### PR TITLE
feat: 🎸 selective decompression of animations

### PR DESCRIPTION
**Pull request by theashraf**
## Type of change

- [ ] Patch: Bug (non-breaking change which fixes an issue)
- [x] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Docs: Documentation updates (if none of the other choices apply)

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have tested my changes on the player-component and React player.

### CODE REVIEW

File: packages/common/src/dotlottie-player.ts [hunk 1]
```diff
@@ -986,11 +982,12 @@ export class DotLottiePlayer {
     let shouldRender = false;
 
     if (revertOptions.includes('activeAnimationId')) {
-      const activeAnimationId = this._manifest?.activeAnimationId;
+      const activeAnimationId = this._getActiveAnimationId();
```
Comment (afsalz) [inline packages/common/src/dotlottie-player.ts:985]: This situation could be a concern. The main purpose of `revertToManifestValues` is to reset specific settings to how they were originally defined in the Manifest. 

For instance, if a user has an `activeAnimationId` in the player props, would the this._getActiveAnimationId(); function retrieve the `activeAnimationId` from player props?

However, what we really need here is to get the `activeAnimationId` from the Manifest, even if the user has set their own `activeAnimationId`.

Comment (afsalz) [reply packages/common/src/dotlottie-player.ts:985]: Suggestion: Adding the following to above this line
```
this._activeAnimationId = this._dotLottieLoader.manifest.activeAnimationId;
```

File: packages/common/src/dotlottie-player.ts [hunk 2]
```diff
@@ -1238,217 +1237,46 @@ export class DotLottiePlayer {
...
-      const srcParsed = DotLottiePlayer.parseSrc(this._src);
+      if (typeof this._src === 'string') {
+        if (isValidLottieString(this._src)) {
+          const validLottieJson = JSON.parse(this._src);
 
-      if (typeof srcParsed === 'string') {
-        const { activeAnimationId, animations, manifest, themes } = await this.getAnimationData(srcParsed);
+          this._dotLottieLoader.loadFromLottieJSON(validLottieJson);
+        } else {
+          const url = new URL(this._src, window.location.href);
```
Comment (afsalz) [inline packages/common/src/dotlottie-player.ts:1246]: We might have SSR issue for react-player.  Can we check if `URL` and `window` exits.

**Review by afsalz — CHANGES_REQUESTED** [review]

File: packages/common/src/dotlottie-player.ts (hunk 1 above)
Comment (theashraf) [reply packages/common/src/dotlottie-player.ts:985]: @afsalz Nice catch!! 👍🏼

File: packages/common/src/dotlottie-player.ts (hunk 2 above)
Comment (afsalz) [reply packages/common/src/dotlottie-player.ts:1246]: works

**Review by afsalz — APPROVED** [review]
Looks good

**Pull request merged**